src/world.rs

```diff
@@ -58,15 +58,17 @@ impl World {
         // AtomicU64 is unsupported on 32-bit MIPS and PPC architectures
         // For compatibility, use Mutex<u64>
         static ID: Mutex<u64> = Mutex::new(1);
-
-        let new_id = ID.lock().checked_add(1).unwrap();
-        *ID.lock() = new_id;
-
-        Self {
-            entities: Entities::default(),
-            archetypes: ArchetypeSet::new(),
-            bundle_to_archetype: HashMap::default(),
-            id: new_id,
+        {
+            let mut lock_result = ID.lock();
+            let id = lock_result.checked_add(1).unwrap();
+            *lock_result = id;
+
+            Self {
+                entities: Entities::default(),
+                archetypes: ArchetypeSet::new(),
+                bundle_to_archetype: HashMap::default(),
+                id,
+            }
         }
```

Comment:
Because this is at the end of a function, wrapping `{}` around it doesn't have any effect. We're also still holding the lock for the duration of the assorted constructors below, when we've already done everything we need. Ideally the lock should be scoped as narrowly as possible, e.g.

```rust
let id = {
    let mut id = ID.lock();
    let next = id.checked_add(1).unwrap();
    *id = next;
    next
};
```